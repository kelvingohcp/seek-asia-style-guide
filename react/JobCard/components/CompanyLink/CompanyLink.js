import React from 'react';
import PropTypes from 'prop-types';
import styles from './CompanyLink.less';
import { getParts } from '../../jobCardHelper.js';
import defaultLink from '../Link/Link';
import { trackLinkType } from '../../JobCard';

export const CompanyLinkPropTypes = PropTypes.shape({
  name: PropTypes.string,
  link: PropTypes.string,
  title: PropTypes.string
});

const CompanyLink = ({ company, keyword = '', LinkComponent = defaultLink, trackLinkClicked }) => {
  const { name, link, title } = company;
  if (!name) {
    return null;
  }

  const companyParts = getParts(name, keyword);

  const companyText = (<span className={link ? styles.companyName : null}>
    {
      companyParts && companyParts.map((part, index) => {
        return (
          <span
            className={part.highlight ? styles.highlight : null}
            key={index}>
            {part.text}
          </span>
        );
      }) || name
    }
  </span>);

  if (link) {
    return (<LinkComponent link={link} className={styles.companyLink} title={title} onClick={() => trackLinkClicked(trackLinkType.company)}>{companyText}</LinkComponent>);
  }

  return companyText;
};

export default CompanyLink;

CompanyLink.propTypes = {
  company: CompanyLinkPropTypes.isRequired,
  keyword: PropTypes.string,
  LinkComponent: PropTypes.func,
  trackLinkClicked: PropTypes.func
};
