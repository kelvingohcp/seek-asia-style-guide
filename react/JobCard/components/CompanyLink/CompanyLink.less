@import (reference) "~seek-asia-style-guide/theme";

.companyName {
  color: @saGrey1;
  font-weight: @sa-semi-bold;
}

.companyLink {
  &:extend(.companyName);
  .link(@saGrey1, true);
}

.highlight {
  font-weight: @sa-bold;
  color: @saBlack;
}