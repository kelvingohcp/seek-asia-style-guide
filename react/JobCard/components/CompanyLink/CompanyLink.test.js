import React from 'react';
import { shallow } from 'enzyme';

import CompanyLink from './CompanyLink';

const defaultCompany = {
  name: 'Seek Asia',
  link: '/jobCard',
  title: 'Jobs at Seek Asia'
};

describe('JobCard - CompanyLink', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<CompanyLink company={defaultCompany} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should bold keyword', () => {
    const wrapper = shallow(<CompanyLink company={defaultCompany} keyword="Seek" />);
    expect(wrapper).toMatchSnapshot();
  });
});
