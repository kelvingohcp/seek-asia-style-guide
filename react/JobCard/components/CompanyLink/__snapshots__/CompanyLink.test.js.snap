// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobCard - CompanyLink should bold keyword 1`] = `
<Link
  className="companyLink"
  link="/jobCard"
  title="Jobs at Seek Asia"
>
  <span
    className="companyName"
  >
    <span
      className="highlight"
      key="0"
    >
      Seek
    </span>
    <span
      className={null}
      key="1"
    >
       Asia
    </span>
  </span>
</Link>
`;

exports[`JobCard - CompanyLink should render correctly 1`] = `
<Link
  className="companyLink"
  link="/jobCard"
  title="Jobs at Seek Asia"
>
  <span
    className="companyName"
  >
    Seek Asia
  </span>
</Link>
`;
